omod-common/src/test/java/org/openmrs/module/webservices/rest/doc/SwaggerSpecificationCreatorTest.java

```diff
@@ -0,0 +1,120 @@
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.webservices.rest.doc;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.GlobalProperty;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.docs.swagger.SwaggerSpecificationCreator;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.api.RestService;
+import org.openmrs.web.test.BaseModuleWebContextSensitiveTest;
+
+import java.sql.*;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SwaggerSpecificationCreatorTest extends BaseModuleWebContextSensitiveTest {
+	
+	Map<String, Integer> beforeCounts;
+	
+	public Map<String, Integer> getRowCounts() throws Exception {
+		Map<String, Integer> ret = new HashMap<String, Integer>();
+		
+		Connection con = this.getConnection();
+		DatabaseMetaData metaData = con.getMetaData();
+		
+		ResultSet rs = metaData.getTables(null, "PUBLIC", "%", null);
+		while (rs.next()) {
+			String tableName = rs.getString(3);
+			
+			Statement stmt = con.createStatement();
+			
+			ResultSet resultSet = stmt.executeQuery("SELECT count(*) AS rowcount FROM " + tableName);
+			
+			resultSet.next();
+			
+			ret.put(tableName, resultSet.getInt("rowcount"));
+			
+			resultSet.close();
+		}
+		
+		return ret;
+	}
+	
+	private void showTableContents(String tableName) throws Exception {
+		Connection con = this.getConnection();
+		Statement stmt = con.createStatement();
+		ResultSet resultSet = stmt.executeQuery("SELECT * FROM " + tableName);
+		
+		ResultSetMetaData metaData = resultSet.getMetaData();
+		int columnCount = metaData.getColumnCount();
+		
+		while (resultSet.next()) {
+			for (int columnIndex = 1; columnIndex <= columnCount; columnIndex++) {
+				Object object = resultSet.getObject(columnIndex);
+				System.out.printf("%s, ", object == null ? "NULL" : object.toString());
+			}
+			System.out.printf("%n");
+		}
+	}
+	
+	@Before
+	public void init() throws Exception {
+		
+		// init REST
+		Context.getService(RestService.class).initialize();
+		
+		Context.getAdministrationService().saveGlobalProperty(
+		    new GlobalProperty(RestConstants.SWAGGER_QUIET_DOCS_GLOBAL_PROPERTY_NAME, "true"));
+		
+		// ensure GP is written to database before we count the rows
+		Context.flushSession();
+		
+		beforeCounts = getRowCounts();
+	}
+	
+	@Test
+	public void checkNoDatabaseChanges() throws Exception {
+		
+		SwaggerSpecificationCreator ssc = new SwaggerSpecificationCreator("/v1/");
+		ssc.BuildJSON();
+		
+		Map<String, Integer> afterCounts = getRowCounts();
+		
+		Assert.assertTrue(ensureCountsEqual(beforeCounts, afterCounts));
+	}
+	
+	private boolean ensureCountsEqual(Map<String, Integer> beforeCounts, Map<String, Integer> afterCounts) throws Exception {
+		if (beforeCounts.size() != afterCounts.size()) {
```

Comment:
You could use junit assertions to check if equal that takes a message too to be printed in case the assertion fails